### PR TITLE
upgrade go modules

### PR DESCRIPTION
**Pull request by haoqing0110**
Upgrade go modules to avoid vulnerable Dependencies 
```
➜  placement git:(br_update) ✗ go mod tidy; go mod vendor; grep -e '#' vendor/modules.txt | awk '{ print $(NF-1), $NF }' | grep -v -e '#' | docker run --rm -i sonatypecommunity/nancy:latest sleuth
Checking for updates...
Already up-to-date.
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Summary                       ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━┫
┃ Audited Dependencies    ┃ 114 ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━┫
┃ Vulnerable Dependencies ┃ 0   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━┛
```
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,24 +2,24 @@ module open-cluster-management.io/placement
...
 require (
-	github.com/onsi/ginkgo v1.14.1
-	github.com/onsi/gomega v1.10.2
-	github.com/openshift/build-machinery-go v0.0.0-20211213093930-7e33a7eb4ce3
-	github.com/openshift/library-go v0.0.0-20210420183610-0e395da73318
-	github.com/spf13/cobra v1.1.3
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.19.0
+	github.com/openshift/build-machinery-go v0.0.0-20220121085309-f94edc2d6874
+	github.com/openshift/library-go v0.0.0-20210420183610-0e395da73318 //upgrade library-go required extra permission to leases.coordination.k8s.io
```
Comment (qiujian16) [inline go.mod:11]: why we cannot upgrade this?

Comment (haoqing0110) [reply go.mod:11]: After upgrading the library-go to latest version, will meet below error. 
```
E0330 03:23:49.443767       1 leaderelection.go:330] error retrieving resource lock open-cluster-management-hub/placement-lock: leases.coordination.k8s.io "placement-lock" is forbidden: User "system:serviceaccount:open-cluster-management-hub:cluster-manager-placement-controller-sa" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "open-cluster-management-hub"
```
And adding below clusterrole can fix this issue.
```
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
  - get
  - list
  - update
```
As the current library-go version doesn't have Vulnerable issues, I would like to use another PR to trace the library-go upgrade and the correct clusterrole we need.

**Pull request merged**